lib/Mojo/IOLoop/ForkCall.pm

```diff
@@ -91,7 +91,7 @@ sub _run {
       waitpid $child, 0;
 
       # attempt to deserialize, emit error and return early
-      my $res = eval { $deserializer->($buffer) };
+      my $res = eval { $deserializer->($buffer) } if $buffer;
```

Comment:
To be honest, I'm not a perl programmer. `$buffer` is initialized with `""`, so a zero-length string. but I did not check, if `Storable::freeze` only produces string values.
I did also not check if `$res` needs to be a zero-length string or can be `$undefined` or anything else.
I guess one would have to dig into the code of `Storable` in order to figure out, what exact data can be expected. Btw, [this](http://blogs.perl.org/users/ovid/2010/10/storable-freeze-versus-nfreeze.html) article suggests to use `nfreeze` instead of `freeze` in order to be faster and more compatible ...
later you send `$res` to the callback `$cb` and the `finish` event of `$self`. They would thus be responsible on what they want to freeze and thaw. So if they don't like to receive a `$undefined` they should make sure they freeze a value they can handle on reception.